### PR TITLE
Add contributing.MD, makefile commands

### PR DESCRIPTION
**Pull request by phalt**
Shuffle around the contributing file to a separate section.
Use makefile commands to make set up easier.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+After cloning this repo, create a [virtualenv](https://virtualenv.pypa.io/en/stable/) and ensure dependencies are installed by running:
+
+```sh
+virtualenv venv
+source venv/bin/activate
+make dev-setup
+git checkout -B YOUR_BRANCH_NAME
+```
+
+You are not ready to start developing
```
Comment (mvanlonden) [inline CONTRIBUTING.md]: Why did I do this then? 😛

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
+# Contributing
+
+## Set up
+
+After cloning this repo, create a [virtualenv](https://virtualenv.pypa.io/en/stable/) and ensure dependencies are installed by running:
```
Comment (mvanlonden) [inline CONTRIBUTING.md]: Could we offer [pipenv](https://docs.pipenv.org/en/latest/) as a lighter weight alternative?

File: CONTRIBUTING.md (hunk 1 above)
Comment (phalt) [reply CONTRIBUTING.md]: Hah I just copy/pasted the original docs

File: CONTRIBUTING.md (hunk 2 above)
Comment (phalt) [reply CONTRIBUTING.md]: @mvanlonden from my personal experience the packaging story of Python is in flux at the moment, I'd rather pick the basic one and when things settle down and a clear winner appears we move over to use it.

Comment (dvndrsn) [reply CONTRIBUTING.md]: Agree on Pipenv. For personal projects, it works fine and has a lot of nice convenience features but there are some things that make it less desirable for cross platform development and especially building libraries.

Comment (ekampf) [reply CONTRIBUTING.md]: "lighter weight" ?  what's wrong with `virtualenv` ?

File: Makefile [hunk 3]
```diff
@@ -9,3 +9,12 @@ docs:
 	pip install -r requirements.txt &&\
 	make html &&\
 	cd -
+
+dev-setup:
+	pip install -e ".[test]"
```
Comment (jkimbo) [inline Makefile]: These commands are going to conflict with https://github.com/graphql-python/graphene/pull/971/files#diff-b67911656ef5d18c4ae36cb6741b7965

**Comment by dvndrsn** [comment]
@jkimbo i was just following @phalt 's lead with those conflicting commands, no worries. Either works.

**Comment by phalt** [comment]
I'm happy to close this in favour of the other one.

**Comment by jkimbo** [comment]
@phalt I think the CONTRIBUTING doc is still really useful from this PR and I didn't mean for you to close it! I was just pointing out that 2 separate PRs have added makefile commands that are doing the same thing (but with different names).

**Pull request closed without merging**